### ISSUE TITLE
bug: Android hostname is not included in cors when changing scheme

### ISSUE THREAD

**Issue by findusl** [ionic-team/capacitor#6936]
### Capacitor Version

```
[warn] The bundledWebRuntime configuration option has been deprecated. Can be safely deleted.
💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 5.4.1
  @capacitor/core: 5.4.1
  @capacitor/android: 5.4.1
  @capacitor/ios: 5.4.1

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 5.4.1
  @capacitor/android: 5.4.1
  @capacitor/core: 5.4.1

[success] Android looking great! 👌
```

### Platform(s)

Android

### Current Behavior

The origin for web requests is `app://` instead of the expected `app://localhost`. This breaks an application if the backend server CORS configuration only allows requests from `app://localhost` and not from `app://`. This only happens on android and only when configuring a custom scheme. It works correctly on iOS with custom scheme.

This is the error message I get on android in our application (backend redacted):
2023-09-25 21:21:15.678 13472-13472 Capacitor/Console       com.example             E  File: app://localhost/#/ - Line 0 - Msg: Access to XMLHttpRequest at '[https://example.com'](https://example.com) from origin 'app://' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### Expected Behavior

Origin is `app://localhost`.

### Code Reproduction

In capacitor.config.ts set the androidScheme

server: {
    androidScheme: 'app',
},

Here is a sample application
https://github.com/findusl/CapacitorAndroidSchemeBug

When starting the android App and connecting with the chrome device debugger, switch to the network tab and hit reload. Pick any request to an external resource, e.g `p-386e91ad.js` and you will note that the Origin request header is `app://`.

When running the same application on iOS, the correct Origin header `app://localhost` is set.

The sample application does not fail, since the accessed resource does not limit CORS. I sadly have not found a publicly available backend to have the request fail.

### Other Technical Details

`npm --version` output: 9.5.1

`node --version` output: v18.16.0

**Comment by jcesarmobile** [comment]
This is a bug on the Android WebView.
It was reported to Google and they closed the issue as "won't fix" and recommended to not use custom schemes on Android and stick with http/https.
